src/proposal.mligo

```diff
@@ -337,18 +337,26 @@ let delete_proposal
     Set.remove (start_date, proposal_key) store.proposal_key_list_sort_by_date
   }
 
+// if proposal is proposed at 0 period_num, it will be expired at 3 period_num
+[@inline]
+let is_proposal_expired (proposal, voting_period, store : proposal * voting_period * storage): bool =
+  let current_period = get_current_period_num(store.start_time, voting_period) in
+  current_period > proposal.period_num + 2n
+
 [@inline]
 let handle_proposal_is_over
-    (config, start_date, proposal_key, store, ops, counter
-      : config * timestamp * proposal_key * storage * operation list * counter
+    (config, start_date, proposal_key, store, ops
+      : config * timestamp * proposal_key * storage * operation list
     )
-    : (operation list * storage * counter) =
+    : (operation list * storage * bool) =
   let proposal = check_if_proposal_exist (proposal_key, store) in
   if    is_voting_period_over (proposal, config.voting_period, store)
```

Comment:
We now calculate the `current_period` twice, in `is_voting_period_over` as well as `is_proposal_expired`, let's calculate it once here and pass it as an argument instead:
```suggestion
  let current_period = get_current_period_num(store.start_time, config.voting_period) in
  if    is_voting_period_over (proposal, current_period)
```